src/main/java/seedu/duke/Parser.java

```diff
@@ -0,0 +1,189 @@
+package seedu.duke;
+
+import storage.Storage;
+import storage.TaskList;
+import task.Deadline;
+import task.Event;
+import task.Task;
+import task.ToDo;
+
+import java.io.IOException;
+import java.text.ParseException;
+import java.text.SimpleDateFormat;
+import java.util.Arrays;
+import java.util.Date;
+
+public class Parser {
+
+    private final String[] cmdTypes = {"find", "list", "delete", "unmark", "mark", "todo", "event", "deadline", "bye"};
+
+    /**
+     * Gets string and checks if command is in cmdTypes
+     * @param input User input
+     * @return String array split into command and the rest of the input
+     * @throws DukeException Command not in cmdTypes
+     */
+    public String[] splitCmd(String input) throws DukeException {
+
+        String[] cmd = input.trim().split(" ", 2);
+        boolean doesExist = Arrays.asList(cmdTypes).contains(cmd[0].trim().toLowerCase());
+
+        if (doesExist) {
+            return cmd;
+        } else {
+            throw new DukeException("Wrong Command Entered");
+        }
+    }
+
+    /**
+     * Check if cmd is the bye command
+     * @param cmd the command that the user input
+     * @return true if is "bye" else false
+     */
+    public boolean checkQuit(String cmd) {
+        return cmd.equalsIgnoreCase(cmdTypes[8]);
+    }
+
+    /**
+     * Parses and runs the command given
+     * @param cmd command type
+     * @param task task to be parsed and run
+     * @param tasks task list of the current program
+     * @param ui ui for displaying messages to the user
+     * @param storage contains the file object for I/O operations to the save file
+     * @throws DukeException A foreseeable error has occurred
+     * @throws IOException Save file error occurred
+     */
+    public void runCmd(String cmd, String task, TaskList tasks, Ui ui, Storage storage)
+            throws DukeException, IOException {
+
+        int idx;
+        Task newTask;
+        String[] parsedTask;
+        switch (cmd.toLowerCase().trim()) {
+
+        case "list":
+            System.out.println(tasks.toString());
+            break;
+
+        case "mark":
```

Comment:
```suggestion
        case "list":
            System.out.println(tasks.toString());
            break;
        case "mark":
```
I am not very sure about this, but the coding standard should be that there shouldn't be a line break after "break;"